src/condor_schedd.V6/qmgmt.h

```diff
@@ -301,44 +307,40 @@ class JobQueueCluster : public JobQueueJob {
 // and assume that they are looking at a JobQueueJob without checking the type
 // so (until we can refactor out this behavior). 
 //
-#if 1 // JobQueueJobSet from JobQueueJob
-class JobQueueJobSet : public JobQueueJob {
-#else
+#ifdef JOB_QUEUE_PAYLOAD_IS_BASE // JobQueueJobSet from JobQueueJob
 class JobQueueJobSet : public JobQueueBase {
+#else
+class JobQueueJobSet : public JobQueueJob {
 #endif
 public:
 	//inherited from JobQueueBase JOB_ID_KEY jid;
 	//inherited from JobQueueBase char entry_type;
 	enum class garbagePolicyEnum { immediateAfterEmpty, delayedAferEmpty };
 
-#if 1 // JobQueueJobSet from JobQueueJob
-public:
-	garbagePolicyEnum garbagePolicy = garbagePolicyEnum::immediateAfterEmpty;
-	unsigned int member_count = 0;
-	LiveJobCounters jobStatusAggregates;
-#else
+#ifdef JOB_QUEUE_PAYLOAD_IS_BASE // JobQueueJobSet from JobQueueJob
 protected:
 	// 3 bytes needed to align the next int
 	char spareA = 0;
 	char spareB = 0;
 	bool dirty = false;
 public:
 	garbagePolicyEnum garbagePolicy = garbagePolicyEnum::immediateAfterEmpty;
-	unsigned int id = 0;
 	unsigned int member_count = 0;
 	struct OwnerInfo * ownerinfo = nullptr;
 	LiveJobCounters jobStatusAggregates;
-	unsigned int Jobset() const { return id; }
+	unsigned int Jobset() const { return (unsigned int)jid.cluster; }
+#else
+public:
+	garbagePolicyEnum garbagePolicy = garbagePolicyEnum::immediateAfterEmpty;
+	unsigned int member_count = 0;
+	LiveJobCounters jobStatusAggregates;
 #endif
 
 public:
-#if 1 // JobQueueJobSet from JobQueueJob
+#ifdef JOB_QUEUE_PAYLOAD_IS_BASE // JobQueueJobSet from JobQueueJob
 	JobQueueJobSet(unsigned int jobset_id)
-		: JobQueueJob(entry_type_jobset)
+		: JobQueueBase(JOB_ID_KEY(jobset_id,JOBSETID_qkey2), entry_type_jobset)
 	{
-		jid.cluster = 0;
-		jid.proc = JOBSETID_to_qkey2(jobset_id);
-		set_id = jobset_id;
 	}
 #else
 	JobQueueJobSet(unsigned int jobset_id)
```

Comment:
This else clause appears to be wrong now (assumes JobQueueJobSet inherits from JobQueueBase instead of JobQueueJob). I'd recommend fixing it or purging the non-JOB_QUEUE_PAYLOAD_IS_BASE code.

Reply:
fixed